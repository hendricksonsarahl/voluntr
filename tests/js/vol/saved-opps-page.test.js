import {
  showOpps,
  showNoOppsMessage,
  renderSavedOpps
} from "../../../src/js/vol/saved-opps-page.js";

let parentElt;
const fullStore = [
  { id: 1, title: "One", event_date: "Thursday, October 26, 2017" },
  { id: 2, title: "Two", event_date: "Thursday, October 26, 2017" },
  { id: 3, title: "Three", event_date: "Thursday, October 26, 2017" }
];
const emptyStore = [];

const setUpDOM = () => {
  document.body.innerHTML = `
    <div id="parent"></div>
  `;

  parentElt = document.getElementById("parent");
};

beforeEach(setUpDOM);

describe("renderSavedOpps", () => {
  it("Should append an H2 element as first child of input element", () => {
    renderSavedOpps(fullStore, parentElt);
    expect(parentElt.childNodes[0].tagName).toBe("H2");
  });

  it("Should append a total of n+1 children when store has size n", () => {
    renderSavedOpps(fullStore, parentElt);
    expect(parentElt.childNodes.length).toBe(4);
  });

  it("Rendered HTML matches snapshot for store with content", () => {
    renderSavedOpps(fullStore, parentElt);
    const renderedHTML = parentElt.innerHTML;
    expect(renderedHTML).toMatchSnapshot();
  });

  it("Rendered HTML matches snapshot for empty store", () => {
    renderSavedOpps(emptyStore, parentElt);
    const renderedHTML = parentElt.innerHTML;
    expect(renderedHTML).toMatchSnapshot();
  });
});
