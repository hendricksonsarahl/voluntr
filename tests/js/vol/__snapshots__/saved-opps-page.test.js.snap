// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSavedOpps Rendered HTML matches snapshot for empty store 1`] = `"<h2>Nothing Saved Yet</h2><p><a href=\\"/opportunities\\">Continue browsing opportunities</a></p>"`;

exports[`renderSavedOpps Rendered HTML matches snapshot for store with content 1`] = `
"<h2>My Saved Posts</h2><div>
    <div class=\\"panel panel-default\\" data-id=\\"1\\">
      <div class=\\"panel-body\\">
        <h4>One
          <br>
          <small>Thursday, October 26, 2017</small>
        </h4>
        <div class=\\"pull-right\\">
          <form action=\\"/match\\" method=\\"post\\">
            <input type=\\"hidden\\" name=\\"oppId\\" value=\\"1\\">
            <button type=\\"submit\\" class=\\"btn btn-primary\\">View&nbsp;
              <span class=\\"glyphicon glyphicon-play\\"></span>
            </button>
            <a class=\\"btn btn-danger remove-button\\" href=\\"#\\" role=\\"button\\">Remove&nbsp;
            <div class=\\"glyphicon glyphicon-remove\\"></div>
            </a>
          </form>
        </div>
      </div>
    </div>
    </div><div>
    <div class=\\"panel panel-default\\" data-id=\\"2\\">
      <div class=\\"panel-body\\">
        <h4>Two
          <br>
          <small>Thursday, October 26, 2017</small>
        </h4>
        <div class=\\"pull-right\\">
          <form action=\\"/match\\" method=\\"post\\">
            <input type=\\"hidden\\" name=\\"oppId\\" value=\\"2\\">
            <button type=\\"submit\\" class=\\"btn btn-primary\\">View&nbsp;
              <span class=\\"glyphicon glyphicon-play\\"></span>
            </button>
            <a class=\\"btn btn-danger remove-button\\" href=\\"#\\" role=\\"button\\">Remove&nbsp;
            <div class=\\"glyphicon glyphicon-remove\\"></div>
            </a>
          </form>
        </div>
      </div>
    </div>
    </div><div>
    <div class=\\"panel panel-default\\" data-id=\\"3\\">
      <div class=\\"panel-body\\">
        <h4>Three
          <br>
          <small>Thursday, October 26, 2017</small>
        </h4>
        <div class=\\"pull-right\\">
          <form action=\\"/match\\" method=\\"post\\">
            <input type=\\"hidden\\" name=\\"oppId\\" value=\\"3\\">
            <button type=\\"submit\\" class=\\"btn btn-primary\\">View&nbsp;
              <span class=\\"glyphicon glyphicon-play\\"></span>
            </button>
            <a class=\\"btn btn-danger remove-button\\" href=\\"#\\" role=\\"button\\">Remove&nbsp;
            <div class=\\"glyphicon glyphicon-remove\\"></div>
            </a>
          </form>
        </div>
      </div>
    </div>
    </div>"
`;
